docs/server/code.md

```diff
@@ -0,0 +1,135 @@
+## Code
+
+This document will cover the design principals that should guide the server side component for the MCP.
+
+The code development is guided as much as possible by the principals behind Onion architecture or Clean architecture:
+
+
+- Dependency Inversion: dependencies should always point in to the layers below and never back out.
+- Dependency injection: defining interfaces that can be swapped out at runtime for different concrete implementations
+- The domain model should always be at the very center and depend on nothing external.
+
+### Layers: 
+
+#### Domain Model
+
+This is the core layer and should depend on none of the other pkgs that make up the application. Our domain model is
+contained in the mobile package as mobile is our core domain the actual model definitions are located in ``` pkg/mobile/types.go``` 
+only core domain types specific to mobile should go here. Other packages can define types as needed for their own needs as long as it 
+is not meant to make up part of the core domain model.
+The core application interfaces are also located in the core mobile package under ```interfaces.go``` these interfaces, define how the core mobile 
+layer expects to be able to interact with external services such as database or web service in order to fulfill its business logic requirements. It is 
+the responsibility of these external layers to fulfill this interface.
+
+
+#### Services (Business logic)
+
+Our services or business logic are contained within the sub packages of the core mobile package. These packages can depend on the core mobile
+package and domain model, but not the other way around, they should also not depend on any database specific implementation or know anything about http. 
```

Comment:
re: "not depend on any database specific implementation or know anything about http",
this is not the case for keycloak & fh-sync-server in the metrics pacakge.
They uses a requester, which deals with requests & responses. And there is setting of the 'Authorization' header in the keycloak one. Should we abstract that away further or move those functions out of those files completely?